### DISCUSSION TITLE
Autocomplete not able to pick an option

### DISCUSSION THREAD

**Discussion by dpalyov** [directus/directus#13808] [Archive]
I am having trouble configuring the autocomplete input. Tried nearly every possible configuration I found on the topic, yet no success.

**My input configuration**

![image](https://user-images.githubusercontent.com/17950490/172867987-94201899-eb49-494a-85f3-b86efd5daa29.png)

**Shape of the result from api**:
```
[
   {name: string, id: string}
]
```

**What is happening is am able to see all options appear in the dropdown, however once selected its as if nothing happens. See screen attached**

https://user-images.githubusercontent.com/17950490/172869566-e9b5a838-8029-4e6b-aeb6-4b59d126d913.mp4

I am running the latest directus version. 

Any help would be appreciated.

**Comment by rijkvanzanten** [comment]
The interface uses the following to generate the list of options:

https://github.com/directus/directus/blob/50e1993bb87364dfde5c6f8e7c898db59ae3743b/app/src/interfaces/input-autocomplete-api/input-autocomplete-api.vue#L104-L121

and the following click handler to emit the value:

https://github.com/directus/directus/blob/50e1993bb87364dfde5c6f8e7c898db59ae3743b/app/src/interfaces/input-autocomplete-api/input-autocomplete-api.vue#L23

The only thing I can think of config wise is that `valuePath` is incorrect 🤔 Are you sure the key is `id` in the API response?

**Reply by dpalyov** [reply]
Hi,
absolutely sure it is ``id``. I have another prop in the response that I can use as value, tried it as well yielding the same result. 
I m getting the above warning in the browser console.

<code> if (Array.isArray(resultsArray) === false) { 
 					// eslint-disable-next-line no-console 
 					console.warn(`Expected results type of array, "${typeof resultsArray}" received`); 
 					return; 
 				} </code>

...but thats normal I suppose, provided  that the ```Results Path``` field is left empty.
I see though that on your click handler it is looping through results array. What happens when results is not iterable? When left blank ```Results path```, ```typeof resultsArray``` is ```string```

**Comment by azrikahar** [comment]
> What is happening is am able to see all options appear in the dropdown, however once selected its as if nothing happens. See screen attached

This is now confirmed to be an unexpected regression due to vue dependency update (as there were no code change in autocomplete interface for months now), and we are tracking this over at #13860.

> I m getting the above warning in the browser console.

With that said, this warning shouldn't appear provided it is an actual array (when Result Path is left blank, this means the response itself should be an array). You initial configuration should already be correct, it's just that the confirmed bug preventing you from actually selecting the options.
